src/routes/transactions/helpers/imitation-transactions.helper.ts

```diff
@@ -0,0 +1,104 @@
+    @Inject(IConfigurationService) configurationService: IConfigurationService,
+  ) {
+    this.vanityAddressChars = configurationService.getOrThrow(
+      'mappings.imitationTransactions.vanityAddressChars',
+    );
+  }
+
+  /**
+   * Filters out outgoing ERC20 transfers that imitate their direct predecessor in
+   * value and have a recipient address that is not the same but matches in vanity.
+   *
+   * @param transactions - list of transactions to filter
+   * @param previousTransaction - transaction to compare last {@link transactions} against
+   *
+   * Note: this only handles singular imitation transfers. It does not handle multiple
+   * imitation transfers in a row, nor does it compare batched multiSend transactions
+   * as the "distance" between those batched and their imitation may not be immediate.
+   */
+  filterOutgoingErc20ImitationTransfers(
+    transactions: Array<TransactionItem>,
+    previousTransaction: TransactionItem | undefined,
+  ): Array<TransactionItem> {
+    // TODO: Instead of filtering, mark transactions so client can display them differently
+    return transactions.filter((item, i, arr) => {
+      // Executed by Safe - cannot be imitation
+      if (item.transaction.executionInfo) {
+        return item;
+      }
+
+      const prevItem = i === arr.length - 1 ? previousTransaction : arr[i + 1];
+
+      // No reference transaction to filter against
+      if (!prevItem) {
+        return item;
+      }
+
+      const txInfo = item.transaction.txInfo;
+      const prevTxInfo = prevItem.transaction.txInfo;
+
+      if (
+        // Only consider transfers...
+        !isTransferTransactionInfo(txInfo) ||
+        !isTransferTransactionInfo(prevTxInfo) ||
+        // ...of ERC20s...
+        !isErc20Transfer(txInfo.transferInfo) ||
+        !isErc20Transfer(prevTxInfo.transferInfo)
+      ) {
+        return item;
+      }
+
+      // ...that are outgoing
+      const isOutgoing = txInfo.direction === TransferDirection.Outgoing;
+      const isPrevOutgoing =
+        prevTxInfo.direction === TransferDirection.Outgoing;
+      if (!isOutgoing || !isPrevOutgoing) {
+        return item;
+      }
+
+      // Imitation transfers are of the same value...
+      const isSameValue =
+        txInfo.transferInfo.value === prevTxInfo.transferInfo.value;
+      if (!isSameValue) {
+        return item;
+      }
+
+      // ...from recipient that has the same vanity but is not the same address
+      return !this.isImitatorAddress(
+        txInfo.recipient.value,
+        prevTxInfo.recipient.value,
+      );
+    });
+  }
+
+  private isImitatorAddress(address1: string, address2: string): boolean {
+    const a1 = address1.toLowerCase();
+    const a2 = address2.toLowerCase();
+
+    // Same address is not an imitation
+    if (a1 === a2) {
+      return false;
+    }
+
+    const isSamePrefix =
+      // Ignore `0x` prefix
+      a1.slice(2, this.vanityAddressChars) ===
+      a2.slice(2, this.vanityAddressChars);
```

Comment:
Just confirming: as this is slicing the `0x` part, only the two following characters are being compared. I mean for example `0x67b450ac7bec2aae5c13ebe4dbaf4f2fa5da0c8d` and `0x67a06f0fecfa567b0c52dc31db17a0b7ddf10252` are considered to be sharing the same prefix (`0x67`). Ommiting the `0x` part when comparing is something intended, right?

Reply:
Good catch, I've added `2` onto `this.vanityAddressChars` in 56f27fb54391f86585816b7df18a2d85265fc5ba.